### PR TITLE
Add Giftcards integration sample

### PR DESCRIPTION
**Pull request by Kwok-he-Chu**

### CODE REVIEW

**Review by gcatanese — CHANGES_REQUESTED** [review]
Good work!

I have added some specific comments (see review) and we should discuss the following:
- simplify the sample app using only the Gift Card component (no drop-in)?
- allow to apply the Gift Card only once?
- is it possible to customise Gift Card component: use `Redeem Gift Card` instead of `Add Gift Card`? use `Gift Card Number` instead of `Card Number`? Change validation message `Security code field is empty`?

File: giftcard-example/wwwroot/js/adyenGiftcardImplementation.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+const redirectResult = urlParams.get('redirectResult');
+
+var remainingAmountToPay = 11000;
+
+// Start checkout experience
+async function startCheckout() {
+    console.info('Start checkout...');
+    try {
+        // Call /paymentMethods endpoint to retrieve payment methods
+        const paymentMethodsResponse = await callServer("/api/paymentMethods");
```
[the remark this answers is not in the document]
Comment (Kwok-he-Chu) [reply giftcard-example/wwwroot/js/adyenGiftcardImplementation.js]: Resolved, decided to mount the individual components in their own containers. Now we no longer get the list of payment methods and resolve this dynamically

When mounting, a developer can use: https://docs.adyen.com/payment-methods/apple-pay/web-component#step-3-mount-the-component

```
applePayComponent
   .isAvailable()
   .then(() => {
       applePayComponent.mount("#applepay-container");
   })
   .catch(e => {
       //Apple Pay is not available
   });
```

File: giftcard-example/readme.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
+# Adyen [Giftcards](https://docs.adyen.com/checkout) Integration Demo
+
+This repository includes examples of PCI-compliant UI integrations for online payments with Adyen. 
+Within this demo app, you'll find a simplified version of an e-commerce website, complete with commented code to highlight key features and concepts of Adyen's API. 
+Check out the underlying code to see how you can integrate Adyen to give your shoppers the option to pay with their preferred payment methods, all in a seamless checkout experience.
+
+![Card checkout demo](wwwroot/images/cardcheckout.gif)
```
[the remark this answers is not in the document]
Comment (Kwok-he-Chu) [reply giftcard-example/readme.md]: I'll add the card animation once everything is approved (the frontend), please review. I'll make the GIF at the end of this week @gcatanese

File: giftcard-example/wwwroot/js/adyenGiftcardImplementation.js [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    });
+    
+    // Append child elements to HTML
+    liElement.appendChild(buttonElement);
+    liElement.appendChild(pPaymentMethodElement);
+    paymentMethodList.appendChild(liElement);
+}
+
+// Hides all payment method buttons after the shopper selected a payment method
+function hideAllPaymentMethodButtons() {
```
[the remark this answers is not in the document]
Comment (Kwok-he-Chu) [reply giftcard-example/wwwroot/js/adyenGiftcardImplementation.js:127]: Added a 'Go back'-button

**Comment by Kwok-he-Chu** [comment]
Merge: https://github.com/adyen-examples/adyen-testing-suite/pull/31 to have the E2E tests run here @gcatanese

**Review by gcatanese — APPROVED** [review]

**Comment by gcatanese** [comment]
@Kwok-he-Chu  Great work. I have committed a minor [edit](https://github.com/adyen-examples/adyen-dotnet-online-payments/pull/72/commits/9d4926d18b48d0539f7a2bc58019e4a37673e0cd) to display an example of GiftCard number (allow devs to copy-paste without leaving the app)

**Pull request merged**